include/deal.II/lac/scalapack.h

```diff
@@ -180,21 +180,84 @@ class ScaLAPACKMatrix : protected TransposeTable<NumberType>
+
+
+
+  /**
+  * Funcion to compute the singular value decomposition (SVD) of an
+  * M-by-N matrix A, optionally computing the left and/or right
+  * singular vectors. The SVD is written as A = U * SIGMA * transpose(V)
+  * where SIGMA is an M-by-N diagonal matrix, @p U is an M-by-M orthogonal matrix,
+  * and @p V is an N-by-N orthogonal matrix. The diagonal elements of SIGMA
+  * are the singular values of A and the columns of U and V are the
+  * corresponding left and right singular vectors, respectively. The
+  * singular values are returned in decreasing order and only the first min(M,N)
+  * columns of U and rows of VT = transpose(V) are computed.
+  * Upon return the content of the matrix is unusable.
+  * The matrix A must have identical block cyclic distribution for the rows and column
+  * If left singular vectors are required matrices A and U
+  * have to be constructed with the same process grid and block cyclic distribution.
+  * If right singular vectors are required matrices A and VT
+  * have to be constructed with the same process grid  and block cyclic distribution.
+  */
+  std::vector<NumberType> compute_SVD(ScaLAPACKMatrix<NumberType> &U,
+                                      ScaLAPACKMatrix<NumberType> &VT,
+                                      const bool left_singluar_vectors=false,
+                                      const bool right_singluar_vectors=false);
+
+
+  /**
+  * Function solves overdetermined or underdetermined real linear
+  * systems involving an M-by-N matrix A, or its transpose, using a QR or LQ factorization of A.
+  *
+  * It is assumed that A has full rank: \f$rank(A) = \min(M,N)\f$.
+  * Upon exit the columns of B contain the solutions and
+  * the following options are supported:
+  * - 1. If transpose==false and \f$M \geq N\f$: least squares solution of overdetermined system
+  *      \f$\min \Vert B - A X\Vert\f$.
+  *
+  *      Upon exit the rows 0 to N-1 contain the least square solution vectors. The residual sum of squares
+  *      for each column is given by the sum of squares of elements N to M-1 in that column
+  *
+  * - 2. If transpose==false and \f$M < N\f$: find minimum norm solutions of underdetermined systems
+  *      \f$A X = B\f$.
+  *
+  *      Upon exit the columns of B contain the minimum norm solution vectors
+  *
+  * - 3. If transpose==true and and \f$M \geq N\f$: find minimum norm solutions of underdetermined system
+  *      \f$ A^\top X = B\f$
+  *
+  *      Upon exit the columns of B contain the minimum norm solution vectors
+  *
+  * - 4. If transpose==true and \f$M < N\f$: least squares solution of overdetermined system
+  *      \f$\min \Vert B - A^\top X\Vert\f$.
+  *
+  *      Upon exit the rows 0 to M-1 contain the least square solution vectors. The residual sum of squares
+  *      for each column is given by the sum of squares of elements M to N-1 in that column
+  * .
+  * If transpose==false B is M x NRHS matrix, otherwise it is NxNRHS.
```

Comment:
what's `NRHS`? I did not see this mentioned

Reply:
Explanation has been added!